packages/server-wallet/src/models/objective.ts

```diff
@@ -148,25 +156,24 @@ export class ObjectiveModel extends Model {
     };
   }
 
-  static async insert(
-    objectiveToBeStored: SupportedObjective & {
-      status: 'pending' | 'approved' | 'rejected' | 'failed' | 'succeeded';
-      waitingFor: WaitingFor;
-    },
-    tx: TransactionOrKnex
-  ): Promise<DBObjective> {
+  static async insert<O extends DBObjective = DBObjective>(
+    objectiveToBeStored: SupportedObjective, // TODO this should be correlated to O
+    preApproved: boolean,
```

Comment:
nit: instead of a boolean we could just accept an `ObjectiveStatus` or a subset of `ObjectiveStatus` (`approved`|`pending`)